ncm-cron: Throw an error if multiple cron jobs are defined with the same name

jrha opened the pull request:
Otherwise this causes the last processed to take precedence.

Resolves #1228.

ned21 commented:
Adding backwards incompatible label as this will break any existing code base where this feature is used which might hamper adoption of the new version. I know at previous workshops that @jouvin described how the use of list and "last entry wins" was a deliberate design decision so I think a bit more discussion is warranted.

jrha commented:
My recollection was that it was a design mistake, but discussion is welcome.

jrha commented:
Also: could easily allow this check to be disabled if needed.

jrha commented:
Discussed at workshop, agreed to define the validation function but leave unbound and leave the decision to bind in a `site_local` template to be site policy.

jrha commented:
Re-based and updated to simply provide a validation function as discussed.
Removed "backwards incompatible" label as the function is not applied by default.

jrha commented:
Fair question, I'd prefer a hard error, can easily make it so.
